src/flow/src/transform/expr.rs

```diff
@@ -33,6 +33,28 @@ use crate::repr::{ColumnType, RelationType};
 use crate::transform::literal::{from_substrait_literal, from_substrait_type};
 use crate::transform::FunctionExtensions;
 
+// TODO: found proper place for this
+/// ref to `arrow_schema::datatype` for type name
+fn typename_to_cdt(name: &str) -> CDT {
+    match name {
```

Comment:
I think it's better to match `name. to_uppercase`.

Reply:
cc @discord9

Reply:
> cc @discord9

Will change it in my next PR, there might be a better way which is to reuse datafusion's logic(which is using sqlparser to parse typename)